vortex-file/src/read/metadata.rs

```diff
@@ -33,74 +51,24 @@ impl<R: VortexReadAt + Unpin> MetadataFetcher<R> {
         root_layout: Box<dyn LayoutReader>,
         layout_cache: Arc<RwLock<LayoutMessageCache>>,
     ) -> Self {
-        Self {
+        let metadata_reader = Coalescer::new(
             input,
             dispatcher,
-            root_layout,
+            Box::new(stream::iter(iter::once(Ok(RowMask::new_valid_between(
+                0, 1,
+            ))))) as _,
+            MetadataMaskReader::new(root_layout),
             layout_cache,
```

Comment:
This suggests broken abstraction. We require some initial value always and we produce this empty range

Reply:
However, it's just simpler to delegate to the existing logic and not have another type. We should factor out common logic from Coalesce for single element operation